source/material_model/visco_plastic.cc

```diff
@@ -248,6 +248,28 @@ namespace aspect
           // Apply strain weakening of the viscous viscosity
           viscosity_pre_yield *= viscous_weakening;
 
+          // For composition spcrust, change from fixed *maximum* viscosity to flow-law viscosity
+          // over specified pressure range (strain weakening could make this lower)
+          if (use_fixed_spcrust_viscosity == true)
+            {
+              if (j == (this->introspection().compositional_index_for_name("spcrust") + 1))
+                {
+                  if (pressure <= spcrust_viscosity_minimum_pressure)
+                    {
+                      viscosity_pre_yield = std::min(maximum_spcrust_viscosity,viscosity_pre_yield);
+                    }
+                  else if ((pressure > spcrust_viscosity_minimum_pressure)
+                           && (pressure < spcrust_viscosity_maximum_pressure))
+                    {
+                      viscosity_pre_yield = std::min(maximum_spcrust_viscosity*
```

Comment:
I think it would be helpful to have a comment briefly writing out and explaining the formula below (it was not intuitive at first glance). 

Also, comparing to the formula for the density change, is there no need for a condition when `pressure >= spcrust_viscosity_maximum_pressure`?

Reply:
I've written a full comment here to explain the calculation. We want to cap the maximum value to the spcrust viscosity at shallow depth and it linearly increases with depth, but we want to allow this crustal value to be lower if the composite rheology predicts that it is lower. There is no need for a condition from pressure above the maximum pressure because in this case the flow-law-defined value (viscosity_pre_yield) is used as is.